compiler/rustc_session/src/session.rs

```diff
@@ -1562,6 +1563,7 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
         "x86_64-unknown-freebsd",
         "x86_64-unknown-linux-gnu",
     ];
+    const HWASAN_SUPPORTED_TARGETS: &[&str] = &["aarch64-unknown-linux-gnu"];
```

Comment:
Does `aarch64-linux-android`/`aarch64-unknown-linux-musl` not support this sanitizer?

On a more general note these should become a target property, rather than be a hardcoded list in the compiler. I'll file a separate issue  for this.

Reply:
HWASan runs on android, so there shouldn't be a problem with `arch64-linux-android` (I haven't had a chance to test it though). I'm not sure about `aarch64-unknown-linux-musl`. @pcc do you know if HWASan runtime is compatible with musl?

Reply:
I added `arch64-linux-android` to the list. As for `aarch64-unknown-linux-musl`, it seems like musl and HWASan don't mix together yet google/sanitizers#1080.